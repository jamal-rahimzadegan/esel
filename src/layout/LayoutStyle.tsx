import React from 'react';
import styled from 'styled-components';
import { Image } from 'components';

export const AppLayout = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  overflow: hidden;
  position: relative;
  background-color: ${({ theme, path }) => {
    if (path === '/') return theme.WHITE;
    return theme[path === '/home' ? 'SECOND_BG' : 'BACKGROUND'];
  }};
`;

export const HeaderContainer = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: 80px;
  position: sticky;
  top: 0;
  background-color: ${({ theme }) => theme.APP};
  z-index: 11;
`;

export const BackBtn = styled(Image)`
  float: left;
  margin: 0 10px;
  cursor: pointer;
  object-fit: contain;
`;

export const Main = styled.main`
  min-height: 100vh;
`;

export const TransitionContainer = styled.div`
  opacity: ${({ transitionType }) => (transitionType === 'in' ? 1 : 0)};
  transition: 400ms;
`;
