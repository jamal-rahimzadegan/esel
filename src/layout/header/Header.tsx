import React from 'react';
import { useRouter } from 'next/router';
import { useHitAction } from 'hooks';
import { ICON_PATH } from 'constant';
import { HyperLink, Image } from 'components/index';
import createAction, { actions } from 'store/actions';
import { BackBtn, HeaderContainer } from '../LayoutStyle';

function Header(): JSX.Element {
  const { asPath: currentRoute, back } = useRouter();
  const hitAction = useHitAction();
  const isOnHomePage = currentRoute === '/home';

  const openDrawerMenu = () => hitAction(createAction(actions.OPEN_DRAWER));

  return (
    <HeaderContainer isOnHomePage={isOnHomePage}>
      <Image
        onClick={openDrawerMenu}
        className="pointer"
        objectFit="contain"
        src={ICON_PATH + 'menu.svg'}
        alt="menu"
        height={28}
        width={28}
      />
      <HyperLink href="/home">
        <Image objectFit="contain" src={ICON_PATH + 'logocrop.svg'} alt="logo" height={35} width={130} />
      </HyperLink>
      {!isOnHomePage ? (
        <BackBtn onClick={back} src={ICON_PATH + 'back.svg'} alt="back" height={40} width={40} />
      ) : (
        <div className="mr-5 ml-4" />
      )}
    </HeaderContainer>
  );
}

export default React.memo(Header);
