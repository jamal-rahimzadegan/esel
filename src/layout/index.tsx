import React from 'react';
import { useRouter } from 'next/router';
import { useSelector } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { getSwipeDirection, toggleTheme } from 'utils';
import { useHitAction } from 'hooks';
import { isDev, lazyLoad, selectorState } from 'constant';
import createAction, { actions } from 'store/actions';
import { GlobalStyle } from 'assets/style/global-style';
import { AppLayout, Main } from './LayoutStyle';
import InternetStatus from './internet-status/InternetStatus';
import { kookie } from 'services/index';

const { Notify, Loader, DrawerMenu, Header, AddToHomeScreen } = {
  Header: lazyLoad(() => import('./header/Header').then((header) => header)),
  Notify: lazyLoad(() => import('components/index').then((module) => module.Notify)),
  Loader: lazyLoad(() => import('components/index').then((module) => module.Loader)),
  DrawerMenu: lazyLoad(() => import('components/index').then((module) => module.DrawerMenu)),
  AddToHomeScreen: lazyLoad(() => import('components/index').then((module) => module.AddToHomeScreen)),
};

interface LayoutProps {
  children: React.ReactNode;
}

export default function Index({ children }: LayoutProps): JSX.Element {
  const { asPath: path } = useRouter();
  const hitAction = useHitAction();
  const {
    theme: { theme },
  } = useSelector(selectorState);

  const showHeaderAndDrawer = !path.includes('user') && path !== '/';

  const handleSwipeEnd = () => {
    if (showHeaderAndDrawer) {
      getSwipeDirection('end').then((dir) => {
        hitAction(createAction(actions[dir === 'left' ? 'OPEN_DRAWER' : 'CLOSE_DRAWER']));
      });
    }
  };

  return (
    <ThemeProvider theme={toggleTheme('light')}>
      <FakeTokenSet />
      <GlobalStyle />
      <InternetStatus />
      {showHeaderAndDrawer ? <Header /> : null}
      <AppLayout
        id="app-layout"
        // onTouchStart={(e) => getSwipeDirection('start', e)}
        // onTouchMove={(e) => getSwipeDirection('move', e)}
        // onTouchEnd={handleSwipeEnd}
        path={path}
      >
        <Main theme={theme}>
          {children}
          {/*<PageTransition>{children}</PageTransition>*/}
        </Main>
        <DrawerMenu />
        <Loader />
        <Notify />
        <AddToHomeScreen />
      </AppLayout>
    </ThemeProvider>
  );
}

function FakeTokenSet() {
  const shouldShow = isDev;
  const tempToken = '';

  const saveToken = () => {
    kookie.set('token', tempToken);
    alert('Token has been set');
  };

  return shouldShow ? <button onClick={saveToken}>set token</button> : null;
}
