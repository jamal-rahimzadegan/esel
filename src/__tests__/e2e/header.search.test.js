import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import { ProviderForTest } from '../setupTests';
import Header from '../../layout/header/Header';
import { BackBtn, HeaderContainer } from '../../layout/LayoutStyle';
import { screen } from '@testing-library/react';
import '../mocks/use-router';

describe('Header ', () => {
  let header;
  beforeEach(() => {
    header = mount(
      <ProviderForTest>
        <Header />
      </ProviderForTest>
    );
  });

  it('renders header', () => {
    expect(header.find(HeaderContainer).prop('isOnHomePage')).toBeDefined();
  });

  it('renders back btn on other pages', () => {
    const back = screen.getByAltText('back');
    expect(back).toHaveProperty('alt', 'back');
  });

  it('does not render back btn on home page', () => {
    expect(header.find(BackBtn).exists()).toBeFalsy();
  });
});
