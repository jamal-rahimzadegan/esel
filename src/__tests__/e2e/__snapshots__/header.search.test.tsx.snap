// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header  should render header 1`] = `
<div
  className="LayoutStyle__HeaderContainer-sc-1txtlos-1 ePVhCm"
>
  <div
    className="d-flex justify-content-end"
  >
    <img
      alt="back"
      className="ImgStyle-sc-1l7md9a-0 dVzkPW mx-1 pointer"
      height={40}
      loading="lazy"
      onError={[Function]}
      radius={null}
      src="/static/icon/back.svg"
      width={40}
    />
  </div>
  <div
    className="d-flex w-100 align-items-center justify-content-between pl-1"
  >
    <div
      className="d-flex align-items-center m-0 p-0 pointer"
      onClick={[Function]}
    >
      <img
        alt="menu"
        className="ImgStyle-sc-1l7md9a-0 dVzkPW"
        height={28}
        loading="lazy"
        onError={[Function]}
        radius={null}
        src="/static/icon/menu.svg"
        width={28}
      />
      <img
        alt="logo"
        className="ImgStyle-sc-1l7md9a-0 dVzkPW"
        height={30}
        loading="lazy"
        onError={[Function]}
        radius={null}
        src="/static/icon/logocrop.svg"
        width={100}
      />
    </div>
    <div
      className="StyledCard-pqrptz-0 fxWBLk col-xl-3 col-lg-3 col-md-4 cl-sm-4 col-7 m-0 p-0 pr-2"
    >
      <div
        className="rounded overflow-hidden shadow-sm"
      >
        <div
          className="InputStyle__InputContainer-t7wfdc-1 cSRctk d-flex align-items-center justify-content-between w-100 undefined"
          height="45px"
        >
          <input
            className="InputStyle__StyledInput-t7wfdc-3 fNAwqa h-100 w-100 px-1"
            name="search"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="جستجو در اشل"
            type="search"
          />
          <div
            className="StyledCard-pqrptz-0 fXnlfF pointer center-items h-100"
            onClick={[Function]}
            width="50px"
          >
            <img
              alt="back"
              className="ImgStyle-sc-1l7md9a-0 kakhi"
              height="30px"
              loading="lazy"
              onError={[Function]}
              radius={null}
              src="/static/icon/search.svg"
              width="30px"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="StyledCard-pqrptz-0 fxWBLk DrawerStyle__SideBarContainer-o7u39i-0 cznMij"
    id="drawer-menu"
  >
    <div
      className="StyledCard-pqrptz-0 fxWBLk DrawerStyle__TriangleDiv-o7u39i-1 gCtwPO"
    />
    <div
      id="drawer"
    >
      <a
        className="HyperLinkStyle__A-lk2z8y-0 gwseho d-flex my-2 align-content-start"
        href="/home"
        onClick={[Function]}
        onMouseEnter={[Function]}
        rel="noopener noreferrer"
        target="_self"
      >
        <img
          alt="home.svg"
          className="ImgStyle-sc-1l7md9a-0 kakhi mx-2"
          height={19}
          loading="lazy"
          onError={[Function]}
          radius={null}
          src="/static/icon/home.svg"
          width={19}
        />
        <p
          className="TxtStyle__StyledText-sc-1cg7ktb-0 cAfWMo"
          onClick={[Function]}
          size="s"
        >
          خانه
        </p>
      </a>
      <a
        className="HyperLinkStyle__A-lk2z8y-0 gwseho d-flex my-2 align-content-start"
        href="/profile"
        onClick={[Function]}
        onMouseEnter={[Function]}
        rel="noopener noreferrer"
        target="_self"
      >
        <img
          alt="user.svg"
          className="ImgStyle-sc-1l7md9a-0 kakhi mx-2"
          height={19}
          loading="lazy"
          onError={[Function]}
          radius={null}
          src="/static/icon/user.svg"
          width={19}
        />
        <p
          className="TxtStyle__StyledText-sc-1cg7ktb-0 cAfWMo"
          onClick={[Function]}
          size="s"
        >
          حساب کاربری
        </p>
      </a>
      <a
        className="HyperLinkStyle__A-lk2z8y-0 gwseho d-flex my-2 align-content-start"
        href="/project/archive"
        onClick={[Function]}
        onMouseEnter={[Function]}
        rel="noopener noreferrer"
        target="_self"
      >
        <img
          alt="archive-done-projects.svg"
          className="ImgStyle-sc-1l7md9a-0 kakhi mx-2"
          height={19}
          loading="lazy"
          onError={[Function]}
          radius={null}
          src="/static/icon/archive-done-projects.svg"
          width={19}
        />
        <p
          className="TxtStyle__StyledText-sc-1cg7ktb-0 cAfWMo"
          onClick={[Function]}
          size="s"
        >
          پروژه های پایان یافته
        </p>
      </a>
      <a
        className="HyperLinkStyle__A-lk2z8y-0 gwseho d-flex my-2 align-content-start"
        href="/calender"
        onClick={[Function]}
        onMouseEnter={[Function]}
        rel="noopener noreferrer"
        target="_self"
      >
        <img
          alt="calender.svg"
          className="ImgStyle-sc-1l7md9a-0 kakhi mx-2"
          height={19}
          loading="lazy"
          onError={[Function]}
          radius={null}
          src="/static/icon/calender.svg"
          width={19}
        />
        <p
          className="TxtStyle__StyledText-sc-1cg7ktb-0 cAfWMo"
          onClick={[Function]}
          size="s"
        >
          تقویم یادداشت
        </p>
      </a>
      <a
        className="HyperLinkStyle__A-lk2z8y-0 gwseho d-flex my-2 align-content-start"
        href="/present"
        onClick={[Function]}
        onMouseEnter={[Function]}
        rel="noopener noreferrer"
        target="_self"
      >
        <img
          alt="logo.svg"
          className="ImgStyle-sc-1l7md9a-0 kakhi mx-2"
          height={19}
          loading="lazy"
          onError={[Function]}
          radius={null}
          src="/static/icon/logo.svg"
          width={19}
        />
        <p
          className="TxtStyle__StyledText-sc-1cg7ktb-0 cAfWMo"
          onClick={[Function]}
          size="s"
        >
          معرفی اپ
        </p>
      </a>
      <a
        className="HyperLinkStyle__A-lk2z8y-0 gwseho d-flex my-2 align-content-start"
        href="/order-ads"
        onClick={[Function]}
        onMouseEnter={[Function]}
        rel="noopener noreferrer"
        target="_self"
      >
        <img
          alt="advertise.svg"
          className="ImgStyle-sc-1l7md9a-0 kakhi mx-2"
          height={19}
          loading="lazy"
          onError={[Function]}
          radius={null}
          src="/static/icon/advertise.svg"
          width={19}
        />
        <p
          className="TxtStyle__StyledText-sc-1cg7ktb-0 cAfWMo"
          onClick={[Function]}
          size="s"
        >
          سفارش آگهی
        </p>
      </a>
      <a
        className="HyperLinkStyle__A-lk2z8y-0 gwseho d-flex my-2 align-content-start"
        href="/telegram-channels"
        onClick={[Function]}
        onMouseEnter={[Function]}
        rel="noopener noreferrer"
        target="_self"
      >
        <img
          alt="cities.svg"
          className="ImgStyle-sc-1l7md9a-0 kakhi mx-2"
          height={19}
          loading="lazy"
          onError={[Function]}
          radius={null}
          src="/static/icon/cities.svg"
          width={19}
        />
        <p
          className="TxtStyle__StyledText-sc-1cg7ktb-0 cAfWMo"
          onClick={[Function]}
          size="s"
        >
          اشل در استان ها
        </p>
      </a>
      <a
        className="HyperLinkStyle__A-lk2z8y-0 gwseho d-flex my-2 align-content-start"
        href="/msg-box"
        onClick={[Function]}
        onMouseEnter={[Function]}
        rel="noopener noreferrer"
        target="_self"
      >
        <img
          alt="envelope.svg"
          className="ImgStyle-sc-1l7md9a-0 kakhi mx-2"
          height={19}
          loading="lazy"
          onError={[Function]}
          radius={null}
          src="/static/icon/envelope.svg"
          width={19}
        />
        <p
          className="TxtStyle__StyledText-sc-1cg7ktb-0 cAfWMo"
          onClick={[Function]}
          size="s"
        >
          ارتباط با ما
        </p>
      </a>
    </div>
  </div>
</div>
`;
